ocaml/xapi/wlb_reports.ml

```diff
@@ -189,8 +189,18 @@ let trim_and_send method_name (start_str, end_str) recv_sock send_sock =
             Buffer.clear recv_buf;
             recv_state := 3
           | [] ->
-            send s;
-            Buffer.clear recv_buf
+            (* send data as much as possiable, just keep the length of `end_str` data to avoid broken tag *)
+            Buffer.clear recv_buf;
+            let s_len = String.length s in
+            let end_str_len = String.length end_str in
+            let send_len = s_len - end_str_len in
+            if send_len > 0 then
+              begin
+                send (String.sub s 0 send_len);
+                Buffer.add_substring recv_buf s send_len end_str_len
```

Comment:
Previously, no data was added to `recv_buf` but now it is. Could you explain why? The same is happening below at `Buffer.add_string recv_buf s` when `|s| <= |end_str|`.